### ISSUE TITLE
Barman doesn't unzip files while recovering the database

### ISSUE THREAD

**Issue by mcondamin** [EnterpriseDB/barman#669]
Hi,

I make backups with Barman using gzip compression.
When I do a recover test, it doesn't unzip the files and copies them as is to the postgresql main directory, then preventing the database from starting.

Here is the command line I use for the recover :
`sudo -u barman barman recover pg02 20220921T144858 /var/lib/postgresql/13/main/ --recovery-staging-path /tmp --remote-ssh-command "ssh postgres@pg02.payxpert.com"`

And here is the output :
```
Starting remote restore for server pg02 using backup 20220921T144858
Destination directory: /var/lib/postgresql/13/main/
Remote command: ssh postgres@pg02.payxpert.com
Copying the base backup.
Copying required WAL segments.
Generating archive status files
Identify dangerous settings in destination directory.

WARNING
The following configuration files have not been saved during backup, hence they have not been restored.
You need to manually restore them in order to start the recovered PostgreSQL instance:

    postgresql.conf
    postgresql.auto.conf
    pg_hba.conf
    pg_ident.conf

Recovery completed (start time: 2022-09-22 14:00:59.117892+00:00, elapsed time: 16 seconds)
Your PostgreSQL server has been successfully prepared for recovery!
```
The server is a Debian 10 and Barman is in version 3.1.0

I can find very little information in the documentation or on the web about recovering a compressed backup, so I don't know if it's a bug or if I'm doing it wrong.

Thank you for you help

**Comment by mikewallace1979** [comment]
It looks like you're doing the right thing to recover a compressed backup so I'm not sure why it isn't working - hopefully we can figure out what's wrong though.

The output suggests that `barman recover` isn't going down the code path for decompressing backups (there should be a message `Staging compressed backup files on the recovery host in: ...` after `Copying the base backup`) so I wonder if the compression field in the backup metadata isn't being set correctly for some reason.

Could you please run `barman diagnose` and post the output (redacting any sensitive information first), or alternatively share the output of `cat /var/lib/barman/pg02/base/20220921T144858/backup.info` and your Barman configuration file(s)?

Also, do you still have logs and output from when the backup was taken? If so it would be useful to see those as well.

**Comment by mcondamin** [comment]
Thank you very much Mike for your help, it was very helpful.

I was looking at the backup.info file and realised that it did not see any compression (none). Then in the configuration file, I saw that I had left the post-backup option that I had added to compress my backup since when I had installed Barman in version 2.X, compression was not native.

The problem came from there ;)
Issue is now fixed, thanks!

Just a small question, in the documentation, for recovering :

```
IMPORTANT: Do not issue a recover command using a target data directory where a PostgreSQL instance is running. In that case, remember to stop it before issuing the recovery. This applies also to tablespace directories.
```

But if I stop the instance where I would like to recover :

```
mc@pg-barman ~$ sudo -u barman barman --log-level DEBUG recover pg02 20220926T145529 /var/lib/postgresql/13/main/ --remote-ssh-command "ssh postgres@pg02.payxpert.com" --target-time="2022-09-26 15:20:00.000" --recovery-staging-path="/tmp"
EXCEPTION: connection to server at "pg02.payxpert.com" (10.10.11.132), port 5432 failed: Connection refused
See log file for more details.
```

Any idea about that ?

**Comment by mikewallace1979** [comment]
This looks like a bug. Barman is trying to connect to the PostgreSQL server to validate that the `backup_compression` options are compatible with the server version, however at recovery time this check is neither necessary nor appropriate.

As a short-term workaround you can comment out `backup_compression` in the Barman configuration and the `barman recover` command will work.

We'll fix this in the next Barman release.

**Comment by mcondamin** [comment]
Thanks in advance for the next release fixing this bug. I will use the workaround in the meantime ;)

**Comment by mikewallace1979** [comment]
Closing since the bug uncovered in https://github.com/EnterpriseDB/barman/issues/669#issuecomment-1258315563 was fixed in release `3.2.0`.
